inference-engine/src/gna_plugin/gna2_model_debug_log.cpp

```diff
@@ -158,4 +162,225 @@ void WriteInputAndOutputTextGNAImpl(const Gna2Model & gnaModel, const std::strin
+                dumpFile << "\tParameter name: " << paramNames[i] << ", value: " << *biasVectorIndex << "\n";
+        }
+        i++;
+    }
+}
+
+static void DumpIntParameter(std::ostream& dumpFile, void** parameters, size_t knownParamCount, const std::vector<std::string> paramNames) {
+    uint32_t* param = reinterpret_cast<uint32_t*>(parameters[0]);
+    if (param != nullptr)
+        dumpFile << "\tParameter name: " << paramNames[0] << ", value: " << *param;
+}
+
+typedef void (*dumpParamaters) (std::ostream&, void**, size_t, const std::vector<std::string>);
+
+static dumpParamaters GetParamDumpFunc(Gna2OperationType type) {
+    static const std::map<Gna2OperationType, dumpParamaters> dumpParamMap = {
+        {Gna2OperationTypeConvolution, DumpConvolutionParameters},
+        {Gna2OperationTypeCopy, DumpCopyParameters},
+        {Gna2OperationTypeFullyConnectedAffine, DumpFCAffineParameters},
+        {Gna2OperationTypeGmm, DumpIntParameter},
+        {Gna2OperationTypeRecurrent, DumpIntParameter}
+    };
+    return dumpParamMap.find(type) != dumpParamMap.end() ? dumpParamMap.find(type)->second : nullptr;
+}
+
+static void DumpPwl(std::ostream& dumpFile, const Gna2Tensor& activation) {
+    const Gna2PwlSegment* const segments = static_cast<Gna2PwlSegment*>(activation.Data);
+    const uint32_t numberOfSegments = activation.Shape.Dimensions[0];
+
+    for (uint32_t k = 0; k < numberOfSegments; k++) {
+        uint32_t scale = ((segments[k].xBase & 3) + 1) * 8;
+        uint64_t factor = 1ULL << scale;
+        int32_t B = segments[k].xBase & 0xfffffffc;
+        double a = static_cast<double>(segments[k].Slope) / factor;
+        double b = static_cast<double>(segments[k].yBase) - ((static_cast<double>(B) * segments[k].Slope) / factor);
+
+        dumpFile << "\n\t\tBase value for input (B) : " << B << "\n";
+        dumpFile << "\t\tBase value for output (b) : " << segments[k].yBase << "\n";
+        dumpFile << "\t\tSegment slope (S): " << segments[k].Slope << "\n";
+        dumpFile << "\t\tShift (scale) : " << scale << "\n";
+        dumpFile << "\t\ty = ax + b:   a = " << a << ", b = " << b;
+        if (segments[k].Slope != 0) {
+            double x0 = static_cast<double>(B) - ((static_cast<double>(segments[k].yBase) * factor) / segments[k].Slope);
+            dumpFile << ", x0 = " << x0;
+        }
+        dumpFile << "\n";
+    }
+}
+
+static void DumpCompoundBias(std::ostream& dumpFile, const Gna2Tensor& activation) {
+    const Gna2CompoundBias* const bias = static_cast<Gna2CompoundBias*>(activation.Data);
+
+    dumpFile << "\t\tBias: " << bias->Bias << ", multiplier: " << unsigned(bias->Multiplier) << "\n";
+}
+
+void DumpGna2Model(const Gna2Model& gnaModel, const std::string dumpFolderNameGNA, bool dumpData) {
+    std::stringstream dumpFileName;
+    uint32_t opsNo = gnaModel.NumberOfOperations;
+
+    dumpFileName << dumpFolderNameGNA << "Gna2ModelDebugDump_" << opsNo << "_layer" << ".txt";
```

Comment:
I would add date and time stamp,
https://en.cppreference.com/w/cpp/io/manip/put_time
in the `ymdHMS` form

Reply:
We had a discussion about the file name during last review, I had this idea, but if I recall correctly we decided no to put the time/date into file name. Should I include it anyway? That would result in storing multiple files. The information about file creation time can be always obtained from the file system information anyway.

Reply:
I would prefer to have timestamp as I would like to avoid overwriting as it is debug capability it should not be a problem

Reply:
Done.